src/workspace/resolver.rs

```diff
@@ -392,6 +394,108 @@ impl WorkspaceResolver {
     }
   }
 
+  /// Prepare the workspace resolver for serialization
+  ///
+  /// The most significant preparation involves converting
+  /// absolute paths into relative (based on `root_dir_url`).
+  /// It also takes care of pre-serializing non-serde internal data.
+  pub fn to_serializable(
+    &self,
+    root_dir_url: BaseUrl,
+  ) -> SerializableWorkspaceResolver {
+    SerializableWorkspaceResolver {
+      import_map: self.maybe_import_map().map(|i| {
+        SerializedWorkspaceResolverImportMap {
+          // TODO: This differs from CLI. Confirm
+          specifier: root_dir_url.make_relative_if_descendant(i.base_url()),
+          json: Cow::Owned(i.to_json()),
+        }
+      }),
+      jsr_pkgs: self
+        .jsr_packages()
+        .map(|pkg| SerializedResolverWorkspaceJsrPackage {
+          relative_base: root_dir_url.make_relative_if_descendant(&pkg.base),
+          name: Cow::Borrowed(&pkg.name),
+          version: Cow::Borrowed(&pkg.version),
+          exports: Cow::Borrowed(&pkg.exports),
+        })
+        .collect(),
+      package_jsons: self
+        .pkg_jsons
+        .iter()
+        .map(|(specifier, pkg_json)| {
+          (
+            // TODO: This differs from CLI. Confirm
+            root_dir_url.make_relative_if_descendant(specifier),
```

Comment:
For reviewer: This differs in that it's using the string used to map the pkg_json within the workspace_resolver, instead of the string within the pkg_json itself.

Reply:
@dsherret just double-checking; do you confirm this is correct?